### PR TITLE
curvefs: support listxattr

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?
#1148 
Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by SeanHai** [comment]
recheck

**Comment by xu-chaojie** [comment]
recheck

**Comment by Wine93** [comment]
recheck

**Comment by Wine93** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

File: curvefs/src/client/fuse_client.cpp [hunk 1]
```diff
@@ -1119,6 +1119,56 @@ CURVEFS_ERROR FuseClient::FuseOpGetXattr(fuse_req_t req, fuse_ino_t ino,
...
+    Inode inode = inodeWrapper->GetInodeLocked();
+    // get xattr key
+    for (const auto &it : inode.xattr()) {
+        // +1 because, the format is key\0key\0
+        *realSize += it.first.length() + 1;
+    }
+
+    // add summary xattr key
+    if (inode.type() == FsFileType::TYPE_DIRECTORY) {
+        *realSize += strlen(XATTRRFILES) + 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/fuse_client.cpp:1143]: if enable SummaryInfo, will it duplicated with xattr  store in inode.xattr?

Comment (SeanHai) [reply curvefs/src/client/fuse_client.cpp:1143]: > if enable SummaryInfo, will it duplicated with xattr stored in inode.xattr?

That will not happen. The curve.dir.r* don't stored in inode xattr and here only return the xattr key.

**Comment by SeanHai** [comment]
recheck

**Review by xu-chaojie — APPROVED** [review]

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

File: curvefs/src/client/curve_fuse_op.cpp [hunk 2]
```diff
@@ -218,6 +218,24 @@ void FuseOpGetXattr(fuse_req_t req, fuse_ino_t ino, const char *name,
...
+    std::memset(buf.get(), 0, size);
+    size_t xattrSize = 0;
+    CURVEFS_ERROR ret = g_ClientInstance->FuseOpListXattr(req, ino, buf.get(),
+                                                          size, &xattrSize);
+    if (ret != CURVEFS_ERROR::OK) {
+        FuseReplyErrByErrCode(req, ret);
+        return;
+    }
+
+    if (size == 0) {
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply curvefs/src/client/curve_fuse_op.cpp:232]: > Why not check if size==0 in the beginning？

The logic of getxattr and listxattr of fuse is:
1. first call the func and the size = 0, you need return the true size.
2. Then call the func with a bigger size the true size again when the fuse get the true size.

so you need get the content size first.

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**